ovs/extensions/services/systemd.py

```diff
@@ -105,6 +105,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
         return params
 
     @staticmethod
+    def regenerate_service(name, client, target_name):
+        """
+        :param name: Template name of the service to add
+        :type name: str
+        :param client: Client on which to add the service
+        :type client: ovs.extensions.generic.sshclient.SSHClient
+        :param target_name: The current service name eg ovs-volumedriver_flash01.service
+        :type target_name: str
+        :return: True if successful
+        :rtype bool
+        """
+        configuration_key = '/ovs/framework/hosts/{0}/services/{1}'.format(System.get_my_machine_id(client),
+                                                                           target_name.replace('ovs-', ''))
+        # If the entry is stored in arakoon, it means the service file was previously made
+        if Configuration.exists(configuration_key):
```

Comment:
If key does not exist, an error should be thrown mentioning `add_service` should be used